### PR TITLE
Add org.zrythm.Zrythm

### PR DESCRIPTION
**Pull request by vchernin**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. @alex-tee
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). (upstream does)
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

Please merge this into flathub-beta only, and give @alex-tee collbaborator access 

Flatpak build seems fairly functional, but app itself is still in alpha so there's definitely some bugs.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by vchernin** [comment]
bot, build org.zrythm.Zrythm

**Comment by alex-tee** [comment]
I am the upstream maintainer/author and I approve of Flathub submission (provided that the `-Dfor_flathub` meson option is used and no patching is done)

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:13]: In general we avoid wayland for plugins as a lot of them don't support it. With VST crashing. Did you test any plugins?

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:25]: This actually doesn't work as $HOME is never expanded. So we end up having to set it in a wrapper script.

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:133]:
```suggestion
```

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:253]: does it really require a more recent Gtk4 than the one in the runtime?

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:378]:
```suggestion
            "buildsystem": "cmake-ninja",
```

?

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:438]: we usuall use `install -d` and post-install in commonl found above `sources`

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:15]: does it really require network?

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:19]: I see no filesystem permission to allow saving files.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.zrythm.Zrythm.json
Comment (alex-tee) [reply org.zrythm.Zrythm.json:253]: yes, some symbols were missing in previous releases. 4.6.0 also has a few important bug fixes

File: org.zrythm.Zrythm.json
Comment (alex-tee) [reply org.zrythm.Zrythm.json:15]: it's only used for checking for new versions at the moment

Comment (hfiguiere) [reply org.zrythm.Zrythm.json:15]:
```suggestion
```

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:25]: I only included `$HOME/.lv2` due to instructions [here](https://github.com/flathub/org.freedesktop.LinuxAudio.BaseExtension/tree/branch/21.08#application-using-audio-plugins). But it does seem to already succesfully pick up plugins from the correct `lv2` directory, so I don't know if it's strictly necessary.

Technically there's a wrapper script already upstream (`zrythm_launch` vs the actual `zrythm` binary). So I guess I could `sed` it, or it could be adjusted somehow upstream.

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:438]: done locally

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:378]: done locally

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:133]: done locally

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:13]: reverted to x11 only, indeed some plugins don't work (and carla is apparently buggy too)

**Comment by vchernin** [comment]
vamp is actually [still broken](https://flathub.org/builds/#/builders/28/builds/9030) sometimes in builds, so my ccache fix did nothing. Will try to figure it out...

File: org.zrythm.Zrythm.json
Comment (vchernin) [reply org.zrythm.Zrythm.json:19]: Some parts of the app don't use GTKFileChooserNative currently, so indeed some filesystem permission is needed, I added filesystem=home. There is also a file browser within the app, so that might not be possible to "fix" without granting lots of permissions.

File: README.md
Comment (hfiguiere) [inline README.md:10]: an end user will never see this.

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [reply org.zrythm.Zrythm.json:26]: that doc is wrong. I know I wrote it. It assumed that variable expansion was done, while it's not.

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:200]:
```suggestion
```

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:255]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: README.md
Comment (vchernin) [reply README.md:10]: Yes, but at least this info exists somewhere for now. Once this is more ready this can be put upstream to docs/manuals somewhere.

File: org.zrythm.Zrythm.json
Comment (alex-tee) [reply org.zrythm.Zrythm.json:26]: some observations/questions:

1. `LXVST_PATH` is not recognized by Zrythm. `VST_PATH` for VST2 and `VST3_PATH` for VST3
2. what happens if the user passes e.g. `LV2_PATH`? are these appended or do they override the user's environment?
3. I can hardcode these paths along with the rest so no messing with environment is needed. can flatpak plugins work with non-flatpak hosts too? or do these only work in flatpak hosts?

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:24]:
```suggestion
        "--env=VST_PATH=/app/extensions/Plugins/lxvst",
```

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [reply org.zrythm.Zrythm.json:26]: 2. that's a question for upstream to answer.

Comment (alex-tee) [reply org.zrythm.Zrythm.json:26]: I meant if you run `LV2_PATH=/mypath flatpak run`, what do these `--env` instructions do? do the re-set environment variables given by the user?

Comment (alex-tee) [reply org.zrythm.Zrythm.json:26]: I hardcoded those paths when `for_flathub` option is passed so the user can still override them with env variables

Comment (vchernin) [reply org.zrythm.Zrythm.json]: Seems to work for vst, vst3, and lv2 plugins now, so resolving this.

File: org.zrythm.Zrythm.json
Comment (hfiguiere) [inline org.zrythm.Zrythm.json:433]: why do you need this?

Comment (alex-tee) [reply org.zrythm.Zrythm.json:433]: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1591

File: org.zrythm.Zrythm.json
Comment (alex-tee) [reply org.zrythm.Zrythm.json:14]: actually this is used for sending bug reports automatically too. preferably the flatpak should include debug symbols for proper backtraces, I think @vchernin is looking into that

File: org.zrythm.Zrythm.json
Comment (alex-tee) [reply org.zrythm.Zrythm.json:433]: actually, 0.3.43 has this bug as well. using 0.3.44 (just released today) would be better since it includes a fix

**Comment by nedrichards** [comment]
bot, build org.zrythm.Zrythm

**Comment by barthalion** [comment]
Nothing particularly blocks the merge here, I'd like a final approval from @alex-tee and @vchernin you're good with how it works.

**Comment by alex-tee** [comment]
I think it's fine for flathub-beta for now. remaining issues can be fixed later

**Pull request closed without merging**
